tests/raftstore_cases/test_split_region.rs

```diff
@@ -197,27 +197,34 @@ fn put_cf_till_size<T: Simulator>(
     let mut len = 0;
     let mut last_len = 0;
     let mut rng = rand::thread_rng();
-    let mut key = vec![];
-    while len < limit {
-        let key_id = range.next().unwrap();
-        let key_str = format!("{:09}", key_id);
-        key = key_str.into_bytes();
+    let mut start = None;
+    let nodes: Vec<u64> = cluster.engines.keys().cloned().collect();
+    loop {
+        let key = format!("{:09}", range.next().unwrap()).into_bytes();
+        if start.is_none() {
+            start = Some(key.clone());
+        }
         let mut value = vec![0; 64];
         rng.fill_bytes(&mut value);
         cluster.must_put_cf(cf, &key, &value);
         // plus 1 for the extra encoding prefix
         len += key.len() as u64 + 1;
         len += value.len() as u64;
-        // Flush memtable to SST periodically, to make approximate size more accurate.
-        if len - last_len >= 1000 {
+        // Approximate size of memtable is inaccurate for small data,
+        // flush memtable to SST periodically, to make approximate size more accurate.
+        let gap = if len >= limit { 200 } else { 1000 };
+        if len - last_len >= gap {
             cluster.must_flush(true);
             last_len = len;
         }
+        if len >= limit {
+            for id in &nodes {
+                if cluster.get_approximate_size(*id, cf, start.as_ref().unwrap(), &key) > limit {
```

Comment:
@huachaohuang why approximate_size is inaccurate in the first place? Since all kvs are distinct, written and flushed. I think the approximate size should not be less than what have been put.